Mathlib/Analysis/CStarAlgebra/Module/Constructions.lean

```diff
@@ -0,0 +1,414 @@
+def equivₗᵢ : C⋆ᵐᵒᵈ A ≃ₗᵢ[ℂ] A where
+  toLinearEquiv := linearEquiv ℂ A
+  norm_map' := norm_equiv
+
+end Self
+
+/-! ## Products of C⋆-modules -/
+
+section Prod
+
+variable {E F : Type*}
+variable [NormedAddCommGroup E] [Module ℂ E] [SMul Aᵐᵒᵖ E]
+variable [NormedAddCommGroup F] [Module ℂ F] [SMul Aᵐᵒᵖ F]
+variable [CStarModule A E] [CStarModule A F]
+
+noncomputable instance : Norm (C⋆ᵐᵒᵈ (E × F)) where
+  norm x := √‖⟪x.1, x.1⟫_A + ⟪x.2, x.2⟫_A‖
+
+lemma prod_norm (x : C⋆ᵐᵒᵈ (E × F)) : ‖x‖ = √‖⟪x.1, x.1⟫_A + ⟪x.2, x.2⟫_A‖ := rfl
+
+lemma prod_norm_sq (x : C⋆ᵐᵒᵈ (E × F)) : ‖x‖ ^ 2 = ‖⟪x.1, x.1⟫_A + ⟪x.2, x.2⟫_A‖ := by
+  simp [prod_norm]
+
+lemma prod_norm_le_norm_add (x : C⋆ᵐᵒᵈ (E × F)) : ‖x‖ ≤ ‖x.1‖ + ‖x.2‖ := by
+  refine abs_le_of_sq_le_sq' ?_ (by positivity) |>.2
+  calc ‖x‖ ^ 2 ≤ ‖⟪x.1, x.1⟫_A‖ + ‖⟪x.2, x.2⟫_A‖ := prod_norm_sq x ▸ norm_add_le _ _
+    _ = ‖x.1‖ ^ 2 + 0 + ‖x.2‖ ^ 2 := by simp [norm_sq_eq]
+    _ ≤ ‖x.1‖ ^ 2 + 2 * ‖x.1‖ * ‖x.2‖ + ‖x.2‖ ^ 2 := by gcongr; positivity
+    _ = (‖x.1‖ + ‖x.2‖) ^ 2 := by ring
+
+variable [StarModule ℂ A] [StarOrderedRing A]
+
+noncomputable instance : CStarModule A (C⋆ᵐᵒᵈ (E × F)) where
+  inner x y := inner x.1 y.1 + inner x.2 y.2
+  inner_add_right {x y z} := by simpa using add_add_add_comm ..
+  inner_self_nonneg := add_nonneg CStarModule.inner_self_nonneg CStarModule.inner_self_nonneg
+  inner_self {x} := by
+    refine ⟨fun h ↦ ?_, fun h ↦ by simp [h, CStarModule.inner_zero_left]⟩
+    apply equiv (E × F) |>.injective
+    ext
+    · refine inner_self.mp <| le_antisymm ?_ inner_self_nonneg
+      exact le_add_of_nonneg_right CStarModule.inner_self_nonneg |>.trans_eq h
+    · refine inner_self.mp <| le_antisymm ?_ inner_self_nonneg
+      exact le_add_of_nonneg_left CStarModule.inner_self_nonneg |>.trans_eq h
+  inner_op_smul_right := by simp [add_mul]
+  inner_smul_right_complex := by simp [smul_add]
+  star_inner x y := by simp
+  norm_eq_sqrt_norm_inner_self {x} := by with_reducible_and_instances rfl
+
+lemma prod_inner (x y : C⋆ᵐᵒᵈ (E × F)) : ⟪x, y⟫_A = ⟪x.1, y.1⟫_A + ⟪x.2, y.2⟫_A := rfl
+
+variable [CStarRing A] [SMulCommClass ℂ A A] [IsScalarTower ℂ A A] [CompleteSpace A]
+
+lemma max_le_prod_norm (x : C⋆ᵐᵒᵈ (E × F)) : max ‖x.1‖ ‖x.2‖ ≤ ‖x‖ := by
+  rw [prod_norm]
+  simp only [equiv_fst, norm_eq_sqrt_norm_inner_self, equiv_snd, max_le_iff, norm_nonneg,
+    Real.sqrt_le_sqrt_iff]
+  constructor
+  all_goals
+    apply norm_le_norm_of_nonneg_of_le
+    all_goals
+      aesop (add safe apply CStarModule.inner_self_nonneg)
+
+lemma norm_equiv_le_norm_prod (x : C⋆ᵐᵒᵈ (E × F)) : ‖equiv (E × F) x‖ ≤ ‖x‖ :=
+  max_le_prod_norm x
+
+section Aux
+
+attribute [-instance] WithCStarModule.instUniformSpace WithCStarModule.instBornology
+attribute [local instance] CStarModule.normedAddCommGroup
+
+open Filter Uniformity Bornology
+
+lemma antilipschitzWith_two_equiv_prod_aux : AntilipschitzWith 2 (equiv (E × F)) :=
+  AddMonoidHomClass.antilipschitz_of_bound (linearEquiv ℂ (E × F)) fun x ↦ by
```

Comment:
I would prefer that we keep it since it's referencing the wrong `MetricSpace` structure on `C⋆ᵐᵒᵈ (E × F)`.

I've gone back and marked all the lemmas in each `Aux` section private for this reason.